### PR TITLE
denso exception error description

### PR DESCRIPTION
**Pull request by paul-chen-medra**
Feature:
- Add back ability to query error description
- Modify bCapException to over ride the what() function in std::exception and bind it
- Add custom exceptions for ExecuteServoTrajectory and change it to throw instead of return hr.
- Add ClearError and bind it
- Clean up some stuff

### CODE REVIEW

File: medra_pybind/src/DensoController.cpp [hunk 1]
```diff
@@ -128,29 +138,29 @@ BCAP_HRESULT DensoController::bCapMotor(bool command) {
...
-        else if (lResult > 0) {
-            std::cout << "Changed to mode 0 ";
-            if (lResult == 1) std::cout << "P-type.\n";
-            if (lResult == 2) std::cout << "J-type.\n";
-            if (lResult == 3) std::cout << "T-type.\n";
-        }
-        else {
-            std::cout << "Released slave mode.\n";
-        }
+        // long lResult;
```
Comment (hashb) [inline medra_pybind/src/DensoController.cpp:141]: delete?

File: medra_pybind/src/bindings.cpp [hunk 2]
```diff
@@ -87,7 +87,7 @@ PYBIND11_MODULE(_medra_bcap, m) {
     .def("SetTcpLoad", &DensoController::SetTcpLoad, py::call_guard<py::gil_scoped_release>()) // Untested
     .def("ChangeTool", &DensoController::ChangeTool, py::call_guard<py::gil_scoped_release>()) // Untested, alternative to SetTcpLoad?
     .def("GetMountingCalib", &DensoController::GetMountingCalib, py::call_guard<py::gil_scoped_release>()) // Untested
-    // .def("GetErrorDescription", &DensoController::GetErrorDescription, py::call_guard<py::gil_scoped_release>()) // Untested
+    .def("GetErrorDescription", &DensoController::GetErrorDescription, py::call_guard<py::gil_scoped_release>()) // Untested
```
Comment (hashb) [inline medra_pybind/src/bindings.cpp:90]: comment is not true anymore?

File: medra_pybind/src/DensoController.cpp (hunk 1 above)
Comment (paul-chen-medra) [reply medra_pybind/src/DensoController.cpp:141]: Moved into a separate function for checking SlvMode

File: medra_pybind/src/bindings.cpp (hunk 2 above)
Comment (paul-chen-medra) [reply medra_pybind/src/bindings.cpp]: fixed

**Review by hashb — APPROVED** [review]
lgtm

**Pull request merged**